jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageWriter.java

```diff
@@ -296,42 +302,59 @@ CachedLayer writeTarLayer(DescriptorDigest diffId, Blob compressedBlob) throws I
   }
 
   /**
-   * Saves the manifest and container configuration for a V2.2 or OCI image.
+   * Saves image metadata (a manifest list and a list of manifest/container configuration pairs) for
+   * an image reference.
    *
    * @param imageReference the image reference to store the metadata for
-   * @param manifestTemplate the manifest
-   * @param containerConfiguration the container configuration
+   * @param manifestList the V2.2 manifest list or OCI image index. Can be null.
+   * @param manifests the V2.2 or OCI manifests
+   * @param containerConfigurations the container configurations
    */
   void writeMetadata(
       ImageReference imageReference,
-      BuildableManifestTemplate manifestTemplate,
-      ContainerConfigurationTemplate containerConfiguration)
+      @Nullable ManifestTemplate manifestList,
```

Comment:
I have been working on changing this to accept `ImageMetadataTemplate` since our last meeting. Still working on it. Because you've already gone through this PR, maybe it's better to avoid further changing this PR a lot. I can do the API refactoring in a follow-up.

Reply:
I decided to change the API in this PR. It enables refactoring the code in a more readable way, I think.

Reply:
Ugh, failed to split the PR. Let me clean this up.

Reply:
Should be good now.